drop-boxes/register-raw-and-mzML/register-raw-and-mzml.py

```diff
@@ -0,0 +1,269 @@
+        out = True
+        for line in mzml:
+            if "<instrumentConfigurationList" in line or 'id="CommonInstrumentParams">' in line:
+                out = False
+            if "</referenceableParamGroup>" in line or "</instrumentConfiguration>" in line:
+                out = True
+            if not out and '<cvParam cvRef="MS"' in line:
+                line = line.split(" ")
+                for token in line:
+                    if "accession=" in token:
+                        accession = token.split('"')[1]
+                break
+        print "accession for "+mzml_path+": "+accession
+        return accession
+
+def parse_timestamp_from_mzml(mzml_path):
+    schema = '{http://psi.hupo.org/ms/mzml}'
+    for event, element in xml.etree.ElementTree.iterparse(mzml_path):
+        if element.tag == schema+'run':
+            xsdDateTime = element.get('startTimeStamp')
+            element.clear()
+            break
+        element.clear() # remove unused xml elements
+    time = None
+    try:
+        time = datetime.datetime.strptime(xsdDateTime, '%Y-%m-%dT%H:%M:%SZ').strftime('%Y-%m-%d %H:%M:%S')
+    except TypeError:
+        print "no startTimeStamp found"
+    return time
+
+def createSimilarMSExperiment(tr, space, project, existing):
+    numberOfExperiments = len(existing)
+    newExpID = '/' + space + '/' + project + '/' + project + 'E' +str(numberOfExperiments)
+    while newExpID in existing:
+        numberOfExperiments += 1 
+        newExpID = '/' + space + '/' + project + '/' + project + 'E' +str(numberOfExperiments)
+    existing.append(newExpID)
+    newExp = tr.createNewExperiment(newExpID, "Q_MS_MEASUREMENT")
+    newExp.setPropertyValue('Q_CURRENT_STATUS', 'FINISHED')
+    newExp.setPropertyValue('Q_ADDITIONAL_INFO', "Automatically created experiment: instrument ID did not fit existing experiment")
+    return newExp
+
+def createSimilarMSSample(tr, space, exp, properties, parents):
+    code = None
+    for p in parents:
+        code = p.split("/")[2]
+    run = 0
+    sampleExists = True
+    newSampleID = None
+    while sampleExists:
+        run += 1
+        newSampleID = '/' + space + '/' + 'MS'+ str(run) + code
+        sampleExists = tr.getSampleForUpdate(newSampleID)
+    newMSSample = tr.createNewSample(newSampleID, "Q_MS_RUN")
+    newMSSample.setParentSampleIdentifiers(parents)
+    newMSSample.setExperiment(exp)
+    newMSSample.setPropertyValue('Q_PROPERTIES', properties)
+    return newMSSample 
+
+
+def handleSampleTracking(barcode):
+    wait_seconds = 1
+    max_attempts = 3
+    for attempt in range(max_attempts):
+        try:
+            SAMPLE_TRACKER.updateSampleStatus(barcode, DATA_AVAILABLE_JSON)
+            break
+        except:
+            print "Updating location for sample "+barcode+" failed on attempt "+str(attempt+1)
+            if attempt < max_attempts -1:
+                time.sleep(wait_seconds)
+                continue
+            else:
+                raise
```

Comment:
Is this necessary? As far as i know we don't show protein based sample type in the sample overview?

Reply:
see above